base/container_iterator.hpp

```diff
@@ -7,14 +7,20 @@ namespace base {
 
 template<typename Container>
 struct ContainerIterator {
-  ContainerIterator(not_null<Container*> container,
-                    typename Container::iterator iterator)
-      : container(container),
-        iterator(iterator) {}
+  using Iterator = typename Container::iterator;
+  ContainerIterator(not_null<Container*> container, Iterator iterator);
+
+  template<typename = std::enable_if_t<
+               std::is_same<Iterator,
+                            decltype(std::declval<Container>().erase(
+                                std::declval<Iterator>()))>::value>>
+  ContainerIterator Erase() const;
 
   not_null<Container*> container;
```

Comment:
Could we make this member private?  It would seem possible now that `Erase` has been added.